### PR TITLE
rpi-base: Added missing HiFiBerry

### PR DESCRIPTION
**Pull request by kluszon**
…Pro, DAC+ DSP, DAC+ HD, Digi Pro.

**- What I did**

Added missing support for HiFiBerry AMP100, AMP3, AMP4 Pro, DAC+ ADC Pro, DAC+ DSP, DAC+ HD, Digi Pro.

**- How I did it**

Add right lines to rpi-base.inc. Changes were tested and works fine with the branch.

Connected also with issue #1168

### CODE REVIEW

**Comment by agherzan** [comment]
Is this a backport from main? If not, why is this kirkstone-specific?

**Comment by kluszon** [comment]
> Is this a backport from main? If not, why is this kirkstone-specific?

It's not a backport, because on **main** branch you don't have it. I'm working on **kirkstone** branch and it's a reason. It could be applied to any branch with kernel 5.10 or newer. I didn't test it with older kernels, but some **dtbo** files can be not available there.

**Comment by kluszon** [comment]
@agherzan Ok then pull request prepared #1287

**Comment by kluszon** [comment]
@agherzan 
Can you do a cherry-pick from 92a9b7a01245507a347c59997a3e5960b0f75ae9 to the Kirkstone branch and close it?

**Comment by agherzan** [comment]
@kluszon Could you refresh this MR by recreating the branch with the cherry-pick of the commit you've prepared for master? This will kick the gears here and run the CI for kirkstone too.l

**Comment by kluszon** [comment]
> @kluszon Could you refresh this MR by recreating the branch with the cherry-pick of the commit you've prepared for master? This will kick the gears here and run the CI for kirkstone too.l

Done

**Review by agherzan — APPROVED** [review]
Looks good.

File: conf/machine/include/rpi-base.inc [hunk 1]
```diff
@@ -25,8 +25,15 @@ RPI_KERNEL_DEVICETREE_OVERLAYS ?= " \
     overlays/gpio-poweroff.dtbo \
     overlays/gpio-shutdown.dtbo \
     overlays/hifiberry-amp.dtbo \
+    overlays/hifiberry-amp100.dtbo \
+    overlays/hifiberry-amp3.dtbo \
+    overlays/hifiberry-amp4pro.dtbo \
     overlays/hifiberry-dac.dtbo \
     overlays/hifiberry-dacplus.dtbo \
+    overlays/hifiberry-dacplusadcpro.dtbo \
```
Comment (agherzan) [inline conf/machine/include/rpi-base.inc:33]: This is not an actual cherry-pick. You squashed another commit into it: https://github.com/agherzan/meta-raspberrypi/commit/fc34bc3b86f9eb42296618a3ceef82f276c472f3

Could you please include both commits for backport?

**Review by agherzan — CHANGES_REQUESTED** [review]

**Comment by agherzan** [comment]
@kluszon the builds fail with:

| make[1]: *** No rule to make target 'arch/arm/boot/dts/overlays/hifiberry-amp4pro.dtbo'.  Stop.

**Comment by kluszon** [comment]
> @kluszon the builds fail with:
> 
> | make[1]: *** No rule to make target 'arch/arm/boot/dts/overlays/hifiberry-amp4pro.dtbo'. Stop.

Your kernel doesn't have the file. I compiled it locally without any problem. What should I do with it?

**Comment by agherzan** [comment]
I wish I had a kernel :)
Joke aside, you're probably building a different version, and not the default kirkstone one. You can see here https://github.com/agherzan/meta-raspberrypi/blob/master/.github/workflows/docker-images/yocto-builder/entrypoint-build.sh how the CI builds the kernel.

**Review by agherzan — CHANGES_REQUESTED** [review]
> The only I can do is remove the problematic amp4pro.

That should work. Could you:
1. rework the removal commit to match the contribution patch format - https://meta-raspberrypi.readthedocs.io/en/latest/contributing.html
2. rebase on latest tip

Thanks.

**Review by agherzan — APPROVED** [review]
Looks good!

**Pull request merged**